dcos/subcommand.py

```diff
@@ -219,40 +225,125 @@ def _write_package_json(pkg):
+    return hasher.hexdigest()
 
-    install_operation = pkg.command_json()
 
-    env_dir = os.path.join(pkg_dir,
-                           constants.DCOS_SUBCOMMAND_VIRTUALENV_SUBDIR)
+def _check_hash(filename, content_hashes):
+    """Validates whether downloaded binary matches expected hash
 
-    if 'pip' in install_operation:
-        _install_with_pip(
-            pkg.name(),
-            env_dir,
-            install_operation['pip'])
+    :param filename: path to binary
+    :type filename: str
+    :param content_hashes: list of hash algorithms/value
+    :type content_hashes: [{"algo": <str>, "value": <str>}]
+    :returns: None if valid hash, else throws exception
+    :rtype: None
+    """
+
+    content_hash = next((contents for contents in content_hashes
+                        if contents.get("algo") == "sha256"),
+                        None)
+    if content_hash:
+        expected_value = content_hash.get("value")
+        actual_value = _hashfile(filename)
+        if expected_value != actual_value:
+            raise DCOSException(
+                "The hash for the downloaded subcommand [{}] "
+                "does not match the expected value [{}]. Aborting...".format(
+                    actual_value, expected_value))
+        else:
+            return
     else:
-        raise DCOSException("Installation methods '{}' not supported".format(
-            install_operation.keys()))
+        raise DCOSException(
+            "Hash algorithm specified is unsupported. "
+            "Please contact the package maintainer. Aborting...")
+
+
+def _get_cli_binary_info(cli_resources):
+    """Find compatible cli binary, if one exists
 
+    :param cli_resources: cli property of resource.json
+    :type resources: {}
+    :returns: {"url": <str>, "kind": <str>, "contentHash": [{}]}
+    :rtype: {} | None
+    """
+
+    if "binaries" in cli_resources:
+        binaries = cli_resources["binaries"]
+        arch = platform.architecture()[0]
+        if arch != "64bit":
+            raise DCOSException(
+                "There is no compatible subcommand for your architecture [{}] "
+                "We only support x86-64. Aborting...".format(arch))
+        system = platform.system().lower()
+        binary = binaries.get(system)
+        if binary is None:
+            raise DCOSException(
+                "There is not compatible subcommand for your system [{}] "
+                "Aborting...".format(binary))
+        elif "x86-64" in binary:
+            return binary["x86-64"]
+
+    raise DCOSException(
+        "The CLI subcommand has unexpected format [{}]. "
+        "Please contact the package maintainer. Aborting...".format(
+            cli_resources))
+
+
+def _install_cli(pkg):
+    """Install subcommand cli
 
-def install(pkg, options):
+    :param pkg: the package to install
+    :type pkg: PackageVersion
+    :rtype: None
+    """
+
+    pkg_dir = _package_dir(pkg.name())
+    with util.path(pkg_dir):
```

Comment:
If you implement the previous comment, you can `with util.remove_path_on_error(_package_dir(pkg.name())) as pkg_dir:`